### PR TITLE
types.db(5): Improve wording and document conventions.

### PR DESCRIPTION
**Pull request by octo**
I just wanted to document that data source names must be unique within a data set (see prometheus/collectd_exporter#78) and ended up rewording most of the manpage.

ChangeLog: types.db(5): Documentation of metric definitions has been improved.

### CODE REVIEW

File: src/types.db.pod [hunk 1]
```diff
@@ -13,21 +13,47 @@ B<collectd>
...
-The format of the data-source specification has been inspired by RRDtool's
-data-source specification. Each data-source is defined by a quadruple made up
-of the data-source name, type, minimal and maximal values, delimited by colons
-(":"): I<ds-name>:I<ds-type>:I<min>:I<max>. I<ds-type> may be either
-B<ABSOLUTE>, B<COUNTER>, B<DERIVE>, or B<GAUGE>. I<min> and I<max> define the
-range of valid values for
-data stored for this data-source. If B<U> is specified for either the min or
-max value, it will be set to unknown, meaning that no range checks will
-happen. See L<rrdcreate(1)> for more details.
+The C<types.db> file contains collectd's metric specifications. Each line
```
Comment (igorpeshansky) [inline src/types.db.pod:16]: How about "metric specifications" -> "metric type specifications"? Also in the next line, "one metric" -> "one metric type". The word "metric" is quite overloaded, so it might confuse some readers.

File: src/types.db.pod [hunk 2]
```diff
@@ -13,21 +13,47 @@ B<collectd>
...
+  total_bytes  value:DERIVE:0:U
+
+The first field defines the name of the data set. By convention, data set names
+use lower-case alphanumeric characters and underscores (C<_>) only. Also by
+convention, if a metric makes sense as a cumulative metric (e.g. C<DERIVE>)
+I<and> a non-cumulative metric (i.e. C<GAUGE>), the cumulative metric gets a
+C<total_> prefix. For example, C<bytes> is a C<GAUGE> and C<total_bytes> is a
+C<DERIVE>.
+
+The second and all following fields define metric values, called "data source".
```
Comment (igorpeshansky) [inline src/types.db.pod:32]: To correspond to "values", should this be "data sources"? Alternatively, you can rephrase this as:

The second and each following field defines a metric value, called "data source".

File: src/types.db.pod [hunk 3]
```diff
@@ -13,21 +13,47 @@ B<collectd>
...
+Data sets with multiple data sources are strongly discouraged. Each field is a
```
Comment (igorpeshansky) [inline src/types.db.pod:33]: [Optional] New data sets with...

File: src/types.db.pod (hunk 2 above)
Comment (igorpeshansky) [inline src/types.db.pod:32]: [Optional] "named metric values"? That way it's clear that the data source name is part of the data source specification.

File: src/types.db.pod [hunk 4]
```diff
@@ -13,21 +13,47 @@ B<collectd>
...
+=item
+
+I<ds-name> is, by convention, a lower-case alphanumeric string. If the data set
+contains a single data source, it is called "value" by convention. Data source
+names must be unique within a data set.
+
+=item
+
+I<ds-type> may be B<DERIVE>, B<GAUGE>, or B<COUNTER>. For historic reasons a
+type called B<ABSOLUTE> is also supported, but its use is strongly discouraged.
```
Comment (igorpeshansky) [inline src/types.db.pod:48]: [Optional] ", and it should not appear in new metric type definitions"?

File: src/types.db.pod [hunk 5]
```diff
@@ -13,21 +13,47 @@ B<collectd>
...
+The first field defines the name of the data set. By convention, data set names
+use lower-case alphanumeric characters and underscores (C<_>) only. Also by
+convention, if a metric makes sense as a cumulative metric (e.g. C<DERIVE>)
+I<and> a non-cumulative metric (i.e. C<GAUGE>), the cumulative metric gets a
+C<total_> prefix. For example, C<bytes> is a C<GAUGE> and C<total_bytes> is a
+C<DERIVE>.
+
+The second and all following fields define metric values, called "data source".
+Data sets with multiple data sources are strongly discouraged. Each field is a
+colon-separated tuple of the data source name, type, minimum and maximum
```
Comment (igorpeshansky) [inline src/types.db.pod:34]: [Optional] "type" -> "value type".

File: src/types.db.pod [hunk 6]
```diff
@@ -13,21 +13,47 @@ B<collectd>
...
+=item
+
+I<ds-type> may be B<DERIVE>, B<GAUGE>, or B<COUNTER>. For historic reasons a
+type called B<ABSOLUTE> is also supported, but its use is strongly discouraged.
+
+=item
+
+I<min> and I<max> define the range of valid values this data source. Either or
+both may be unbounded, which is specified by providing C<U> instead of a
+number.
```
Comment (igorpeshansky) [inline src/types.db.pod:54]: [Optional] Should we add "The type of the min and max values corresponds to the value type (double for GAUGE, int64 for DERIVE and COUNTER)"?

File: src/types.db.pod (hunk 2 above)
Comment (octo) [reply src/types.db.pod]: Used the rephrased version, thanks!

File: src/types.db.pod (hunk 4 above)
Comment (octo) [reply src/types.db.pod]: Feels redundant, but I'm not the native speaker ;) Done.

File: src/types.db.pod (hunk 6 above)
Comment (octo) [reply src/types.db.pod]: Good point, for `DERIVE` and `COUNTER`, min and max refer to the rate, not the raw value. Added a note about this.

File: src/types.db.pod [hunk 7]
```diff
@@ -13,21 +13,49 @@ B<collectd>
...
+=item
+
+I<ds-name> is, by convention, a lower-case alphanumeric string. If the data set
+contains a single data source, it is called "value" by convention. Data source
+names must be unique within a data set.
+
+=item
+
+I<ds-type> may be B<DERIVE>, B<GAUGE>, or B<COUNTER>. For historic reasons a
+type called B<ABSOLUTE> is also supported, but its use is strongly discouraged
```
Comment (igorpeshansky) [inline src/types.db.pod:48]: Nit: missing comma at the end of this line.

File: src/types.db.pod [hunk 8]
```diff
@@ -13,21 +13,49 @@ B<collectd>
...
+consists of two or more fields delimited by spaces and/or horizontal tabs.
+
+For example, the following defines two data sets, "bytes" and "total_bytes".
+
+  bytes        value:GAUGE:0:U
+  total_bytes  value:DERIVE:0:U
+
+The first field defines the name of the data set. By convention, data set names
+use lower-case alphanumeric characters and underscores (C<_>) only. Also by
+convention, if a metric makes sense as a cumulative metric (e.g. C<DERIVE>)
```
Comment (igorpeshansky) [inline src/types.db.pod:27]: I'd add a "both" between "makes sense" and "as a" (or between "as" and "a" — your choice, both seem grammatically correct).

File: src/types.db.pod (hunk 7 above)
Comment (octo) [reply src/types.db.pod]: Thanks! A commas in English are very different from commas in German … :\

File: src/types.db.pod (hunk 8 above)
Comment (octo) [reply src/types.db.pod]: Done.

**Comment by octo** [comment]
Thanks Igor!
